tests/bdd/steps/geometry_factory.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of osm2pgsql (https://osm2pgsql.org/).
+#
+# Copyright (C) 2022 by the osm2pgsql developer community.
+# For a full list of authors see the git log.
+
+class GeometryFactory:
+    """ Provides functions to create geometries from data grids.
+    """
+
+    def __init__(self):
+        self.grid = {}
+
+
+    def parse_geometry(self, geom, scene):
+        """ Create a WKT SQL term for the given geometry.
+            The function understands the following formats:
+
+              <P>
+                 Point geometry
+              <P>,...,<P>
+                 Line geometry
+              (<P>,...,<P>)
+                 Polygon geometry
+
+           <P> may either be a coordinate of the form '<x> <y>' or a single
+           number. In the latter case it must refer to a point in
+           a previously defined grid.
+        """
+        if geom.find(',') < 0:
+            out = "POINT({})".format(self.mk_wkt_point(geom))
+        elif geom.find('(') < 0:
+            out = "LINESTRING({})".format(self.mk_wkt_points(geom))
+        else:
+            out = "POLYGON(({}))".format(self.mk_wkt_points(geom.strip('() ')))
+
+        return "ST_SetSRID('{}'::geometry, 4326)".format(out)
```

Comment:
Use EWKT format `SRID=4326;POLYGON((0 0,0 1,1 1,1 0,0 0))` instead of the `ST_SetSRID()`?